src/main/java/at/unisalzburg/apted/node/Node.java

```diff
@@ -37,24 +38,24 @@
   /**
    * Information associated to and stored at each node. This can be anything
    * and depends on the application, for example, string label, key-value pair,
-   * vector of values, etc.
+   * list of values, etc.
    */
   private D nodeData;
 
   /**
    * Array of pointers to this node's children. The order of children is
    * significant due to the definition of ordered trees.
    */
-  private Vector<Node<D>> children;
+  private List<Node<D>> children;
 
   /**
-   * Constructs a new node with the passed node data and an empty vector of
+   * Constructs a new node with the passed node data and an empty list of
    * children.
    *
    * @param nodeData instance of node data (node label).
    */
   public Node(D nodeData) {
-    this.children = new Vector<Node<D>>();
+    this.children = new ArrayList<>();
```

Comment:
Shouldn't it be 'ArrayList<Node<D>>'?

Reply:
No, not necessarily. Java infers this type info from `this.children`. Declaring it in both places is redundant.